docs/using-react-redux/connect-dispatching-actions-with-mapDispatchToProps.md

```diff
@@ -0,0 +1,419 @@
+
+// binds on `props` change
+const mapDispatchToProps = (dispatch, ownProps) => {
+  toggleTodo: () => dispatch(toggleTodo(ownProps.todoId));
+};
+```
+
+### Return
+
+Your `mapDispatchToProps` function should return a plain object:
+
+- Each field in the object should be a function, calling which should automatically dispatch the action
+- If you use action creators ( as oppose to plain object actions ) inside `dispatch`, it is a convention to simply name the field key the same name as the action creator:
+
+```js
+const increment = () => ({ type: "INCREMENT" });
+const decrement = () => ({ type: "DECREMENT" });
+const reset = () => ({ type: "RESET" });
+
+const mapDispatchToProps = dispatch => {
+  return {
+    // dispatching actions returned by action creators
+    increment: () => dispatch(increment()),
+    decrement: () => dispatch(decrement()),
+    reset: () => dispatch(reset())
+  };
+};
+```
+
+The return of the `mapDispatchToProps` function will be merged to your connected component as props. You may call them directly to dispatch its action.
+
+```js
+function Counter({ count, increment, decrement, reset }) {
+  return (
+    <div>
+      <button onClick={decrement}>-</button>
+      <span>{count}</span>
+      <button onClick={increment}>+</button>
+      <button onClick={reset}>reset</button>
+    </div>
+  );
+}
+```
+
+(Full code of the Counter example is [in this CodeSandbox](https://codesandbox.io/s/yv6kqo1yw9))
+
+### Defining the `mapDispatchToProps` Function with `bindActionCreators`
+
+Wrapping these functions by hand is tedious, so Redux provides a function to simplify that.
+
+> `bindActionCreators` turns an object whose values are [action creators](https://redux.js.org/glossary#action-creator), into an object with the same keys, but with every action creator wrapped into a [`dispatch`](https://redux.js.org/api/store#dispatch) call so they may be invoked directly. See [Redux Docs on `bindActionCreators`](http://redux.js.org/docs/api/bindActionCreators.html)
+
+`bindActionCreators` accepts two parameters:
+
+1. A **`function`** (an action creator) or an **`object`** (each field an action creator)
+2. `dispatch`
+
+```js
+import { bindActionCreators } from "redux";
+
+const increment = () => ({ type: "INCREMENT" });
+const decrement = () => ({ type: "DECREMENT" });
+const reset = () => ({ type: "RESET" });
+
+// binding an action creator
+// returns () => dispatch(increment())
+bindActionCreators(increment, dispatch);
+
+// binding an object full of action creators
+bindActionCreators(
+  {
+    increment,
+    decrement,
+    reset
+  },
+  dispatch
+);
+// returns
+// {
+//   increment: () => dispatch(increment()),
+//   decrement: () => dispatch(decrement()),
+//   reset: () => dispatch(reset()),
+// }
+```
+
+To use `bindActionCreators` in our `mapDispatchToProps` function:
+
+```js
+import { bindActionCreators } from "redux";
+// ...
+
+function mapDispatchToProps(dispatch) {
+  return bindActionCreators(
+    {
```

Comment:
I think we can collapse this object too, and leave off the comments - that point was made already.